include/up-cpp/utils/CallbackConnection.h

```diff
@@ -234,6 +234,18 @@ struct BadConnection : public std::runtime_error {
 	    : std::runtime_error(std::forward<Args>(args)...) {}
 };
 
+/// @brief Thrown if an empty std::function parameter was received
+///
+/// A std::function can be empty. When an empty function is invoked, it will
+/// throw std::bad_function_call. We can check earlier by casting the function
+/// to a boolean. If the check fails, EmptyFunctionObject is thrown. This makes
+/// the error appear earlier without waiting for invokation to occur.
```

Comment:
I wonder... it's called [`std::invoke`](https://en.cppreference.com/w/cpp/utility/functional/invoke), but on the other hand the normal English here would be invocation. I could go either way if in C++ it's standard to call this "invokation".
```suggestion
/// the error appear earlier without waiting for invocation to occur.
```